contracts/src/v0.8/automation/dev/v2_3/AutomationRegistry2_3.sol

```diff
@@ -413,6 +367,59 @@ contract AutomationRegistry2_3 is AutomationRegistryBase2_3, OCR2Abstract, Chain
     );
   }
 
+  function _updateTransmitters(address[] memory signers, address[] memory transmitters) internal {
+    // move all pooled payments out of the pool to each transmitter's balance
+    for (uint256 i = 0; i < s_transmittersList.length; i++) {
+      _updateTransmitterBalanceFromPool(
+        s_transmittersList[i],
+        s_hotVars.totalPremium,
+        uint96(s_transmittersList.length)
+      );
+    }
+
+    // remove any old signer/transmitter addresses
+    address signerAddress;
+    address transmitterAddress;
+    PayoutMode mode = s_payoutMode;
+    for (uint256 i = 0; i < s_transmittersList.length; i++) {
+      signerAddress = s_signersList[i];
+      transmitterAddress = s_transmittersList[i];
+      delete s_signers[signerAddress];
+      // Do not delete the whole transmitter struct as it has balance information stored
+      s_transmitters[transmitterAddress].active = false;
+      if (mode == PayoutMode.OFF_CHAIN) {
+        s_deactivatedTransmitters.add(transmitterAddress);
```

Comment:
should we only add them to this list if their balance > 0?

Reply:
done